source/lac/petsc_parallel_sparse_matrix.cc

```diff
@@ -890,6 +890,48 @@ namespace PETScWrappers
       return u*tmp;
     }
 
+    IndexSet
+    SparseMatrix::locally_owned_domain_indices () const
+    {
+      PetscInt n_rows, n_cols, min, max;
+      PetscErrorCode ierr;
+      IS *rows = nullptr;
+      IS *cols = nullptr;
+
+      ierr = MatGetSize (matrix, &n_rows, &n_cols);
+      AssertThrow (ierr == 0, ExcPETScError(ierr));
+      ierr = MatGetOwnershipIS(matrix, rows, cols);
+      AssertThrow (ierr == 0, ExcPETScError(ierr));
+      ierr = ISGetMinMax(*rows, &min, &max);
+      AssertThrow (ierr == 0, ExcPETScError(ierr));
+
+      IndexSet locally_owned_domain_indices(n_rows);
+      locally_owned_domain_indices.add_range(min, max);
```

Comment:
I agree, i was just thinking if it is possible to double check that in case at some point in future PETSc will become more flexible and this part of the code will break due to the underlying assumption. But you are right, it is certainly not important for now.